backend/internal/database/img.go

```diff
@@ -0,0 +1,75 @@
+package database
+
+import (
+	"bytes"
+	"context"
+	"image"
+
+	_ "image/jpeg"
+	_ "image/png"
+
+	"github.com/buckket/go-blurhash"
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/pkg/errors"
+)
+
+type ImgDatabase struct {
+	Conn *pgxpool.Pool
+}
+
+/*
+ * Gets an image from the image table given the id. If image exists, returns the binary data of the image
+ */
+func (db *ImgDatabase) GetImage(ctx context.Context, id string) (*[]byte, *string, error) {
+	var img []byte
+	var ctype string
+	var query string = "SELECT img, mime_type FROM image WHERE id=$1"
+
+	err := db.Conn.QueryRow(ctx, query, id).Scan(&img, &ctype)
+
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "error in SELECT")
+	}
+
+	return &img, &ctype, nil
+
+}
+
+/*
+ * Puts an image into the image table and returns id (generated using blurhash) and
+ * a boolean specifying if an image was entered into the db. If the image
+ * already exists, returns hash of pre-existing image
```

Comment:
I think it would be better for the controller to do a `GetImage` to see if the image already exists, rather than sending a few MB (potentially) to the database only to find out the file already exists.